test/e2e/specs/editor/various/rich-text.spec.js

```diff
@@ -0,0 +1,826 @@
+		await pageUtils.pressKeys( 'primary+b' );
+		await page.keyboard.type( '1' );
+		await pageUtils.pressKeys( 'primary+b' );
+		await page.keyboard.type( ' 2' );
+		await pageUtils.pressKeys( 'shift+ArrowLeft' );
+		await pageUtils.pressKeys( 'primary+b' );
+
+		const count = await editor.canvas.evaluate(
+			() =>
+				document.querySelectorAll( '*[data-rich-text-format-boundary]' )
+					.length
+		);
+		expect( count ).toBe( 1 );
+	} );
+
+	test( 'should return focus when pressing formatting button', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'Some ' );
+		await editor.clickBlockToolbarButton( 'Bold' );
+		await page.keyboard.type( 'bold' );
+		await editor.clickBlockToolbarButton( 'Bold' );
+		await page.keyboard.type( '.' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: { content: 'Some <strong>bold</strong>.' },
+			},
+		] );
+	} );
+
+	test( 'should transform backtick to code', async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'A `backtick`' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: { content: 'A <code>backtick</code>' },
+			},
+		] );
+
+		await pageUtils.pressKeys( 'primary+z' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [
+			{ name: 'core/paragraph' },
+		] );
+	} );
+
+	test( 'should undo backtick transform with backspace', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '`a`' );
+		await page.keyboard.press( 'Backspace' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: { content: '`a`' },
+			},
+		] );
+	} );
+
+	test( 'should not undo backtick transform with backspace after typing', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '`a`' );
+		await page.keyboard.type( 'b' );
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.press( 'Backspace' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [] );
+	} );
+
+	test( 'should not undo backtick transform with backspace after selection change', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '`a`' );
+		await page.evaluate( () => new Promise( window.requestIdleCallback ) );
+		// Move inside format boundary.
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.press( 'Backspace' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [] );
```

Comment:
I might be missing something here, but how does this ensure the backtick transform is not undone?

Reply:
Because it's removed `a` and didn't restore `` `a` ``?

Reply:
So the test is for ensuring backspace works as expected inside the inline code boundary? Was there a regression where hitting backspace inside that boundary restored the string as well?

Reply:
No, this test is making sure that the transform is not undone after a selection change. Instead of undoing the transform, it should remove the character (default behaviour).